### PR TITLE
[FIX] - 8.0 issue 152 Earned Value Analysis

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
This PR resolves the division by zero issue in Earned Value Analysis Reports.

Please review @elicoidal

### CODE REVIEW

File: business_requirement_earned_value/report/br_earned_value_report.py [hunk 1]
```diff
@@ -94,6 +94,8 @@ def _select(self):
                                    project_task pt
                                WHERE
                                    pt.business_requirement_id = br.id
+                                   AND
+                                   SUM(res.unit_price) > 0
```
Comment (elicoidal) [inline business_requirement_earned_value/report/br_earned_value_report.py]: Doing so we would exclude any line with price = 0.
I am not sure I want to do this (The sum of quantity would be removed as well)

**Comment by serpentcs-dev1** [comment]
@elicoidal  
Please review it

File: business_requirement_earned_value/report/br_earned_value_report.py [hunk 2]
```diff
@@ -88,15 +88,29 @@ def _select(self):
                          WHERE
                              pt.business_requirement_id = br.id) > 0
                     THEN
-                        (abs(((SELECT
+                        CASE
+                            WHEN 
+                              SUM(res.unit_price) > 0
```
Comment (elicoidal) [inline business_requirement_earned_value/report/br_earned_value_report.py]: It should be <> 0
ELSE 0

**Comment by serpentcs-dev1** [comment]
@elicoidal
I have improved changes as per comment please review it

**Review by elicoidal — APPROVED** [review]
@seb-elico @victormartinelicocorp please check

File: business_requirement_earned_value/report/br_earned_value_report.py [hunk 3]
```diff
@@ -88,15 +88,22 @@ def _select(self):
                          WHERE
                              pt.business_requirement_id = br.id) > 0
                     THEN
-                        (abs(((SELECT
+                        CASE
+                            WHEN 
+                              SUM(res.unit_price) <> 0
+                            THEN
+                                abs(((SELECT
                                    SUM(pt.effective_hours)
```
Comment (seb-elico) [inline business_requirement_earned_value/report/br_earned_value_report.py]: Could you please fix the indentation of these lines?

**Review by seb-elico — CHANGES_REQUESTED** [review]

File: business_requirement_earned_value/report/br_earned_value_report.py (hunk 3 above)
Comment (serpentcs-dev1) [reply business_requirement_earned_value/report/br_earned_value_report.py]: Sure will do the needful.

**Comment by dreispt** [comment]
Testes are failing.

**Comment by serpentcs-dev1** [comment]
@dreispt The code to resolve the tests are already passed in #139

Waiting for it to get merged :)

**Comment by elicoidal** [comment]
Travis is in pain... cannot finish a test. Will have to wait for a while

**Comment by dreispt** [comment]
You need to rebase to have the changes from the merged PR available here.

**Comment by serpentcs-dev1** [comment]
Rebased, travis shows green. Good to go now.

**Comment by elicoidal** [comment]
@darshan-serpent I am not sure there is not a problem actually. A lot of changes that you are proposing in this PR are out of scope and / or already implemented in other PR.
Are you sure you have rebased properly?

**Comment by serpentcs-dev1** [comment]
@elicoidal Yeah you were right. So I think it is fine now.

**Review by elicoidal — APPROVED** [review]

**Pull request merged**